code/game/machinery/doors/brigdoors.dm

```diff
@@ -109,13 +71,130 @@
 
 /obj/machinery/door_timer/Destroy()
 	QDEL_NULL(Radio)
-	targets.Cut()
 	prisoner = null
+	targets.Cut()
 	return ..()
 
-//Main door timer loop, if it's timing and time is >0 reduce time by 1.
-// if it's less than 0, open door, reset timer
-// update the door_timer window and the icon
+///////////////////////////////////////////////////////CELL LOGS/////////////////////////////////////////////////
+
+/obj/machinery/door_timer/proc/print_report(mob/user) //Printing Cell Logs
+	for(var/obj/machinery/computer/prisoner/C in prisoncomputer_list)
+		P = new /obj/item/weapon/paper(C.loc)
+		P.name = "[id] log - [detaineename] [worldtime2text()]"
+		P.info =  "<center>[station_name()] - Security Department</center>"
+		P.info += {"<center><b><h3>[id] - Brig record</h3></b></center>
+					<center><b>Admission data:</b></small></center><br>
+					<small><b>Log generated at:</b>		[worldtime2text()]<br>
+					<b>Detainee:</b>		[detaineename]<br>
+					<b>Duration:</b>		[timetoset/10] seconds<br>
+					<b>Charge(s):</b>		[detaineecrimes]<br>
+					<b>Arresting Officer:</b>		[user.name]<br></small>
+					<br>This log file was generated automatically upon activation of a cell timer.<br><hr>"}
+
+		playsound(C.loc, "sound/goonstation/machines/printer_dotmatrix.ogg", 50, 1)
+	cell_logs.Add(P)
+	return 1
+
+///////////////////////////////////////////////RECORD UPDATING/////////////////////////////////////////////////////
+
+/obj/machinery/door_timer/proc/update_record(mob/user)
+	var/datum/data/record/R = find_security_record("name", detaineename)
+	Radio.autosay("Detainee: [detaineename] has been incarcerated for; [timetoset/10] seconds for the charges of; [detaineecrimes]\
+	Arresting Officer: [user.name].[R ? "" : " Detainee record not found, manual record update required."]", name, "Security", list(z))
+
+	if(R)
+		prisoner = R
+		R.fields["criminal"] = "Incarcerated"
+		var/mob/living/carbon/human/M = user
+		var/rank = "UNKNOWN RANK"
+		if(istype(M) && M.wear_id)
+			var/obj/item/weapon/card/id/I = M.wear_id
+			rank = I.assignment
+		R.fields["comments"] = list()
+		R.fields["comments"] += "Autogenerated by [name] on [current_date_string] [worldtime2text()]<BR>Sentenced to [timetoset/10] seconds for the charges of \"[detaineecrimes]\" by [rank] [user.name]."
+		update_all_mob_security_hud()
+
+	return 1
+ ///////////////////////////////////////////////////////CRIMES AND MODIFIERS/////////////////////////////////////////////////
+
+/obj/machinery/door_timer/proc/clearstringandtime() //Updating comitted crimes
+	detaineecrimes = ""//Empty the string list so we can fill it again
+	suggestedbrigtime = 0
+	timetoset = 0
+	for(var/datum/spacelaw/C in comittedcrimes)
+		detaineecrimes += "[C.name]<br>"

```

Comment:
This formats weird on the radio.
![image](https://user-images.githubusercontent.com/6334647/27255638-c163ba22-5367-11e7-8b5d-6c31e3fd8cec.png)